C/primitive/elements.c

```diff
@@ -0,0 +1,424 @@
+#include <simplicity/elements.h>
+
+#include <stdlib.h>
+#include <stdalign.h>
+#include <string.h>
+#include "elements/primitive.h"
+#include "../deserialize.h"
+#include "../eval.h"
+#include "../sha256.h"
+#include "../typeInference.h"
+
+#define PADDING(alignType, allocated) ((alignof(alignType) - (allocated) % alignof(alignType)) % alignof(alignType))
+
+/* Add a 256-bit hash to be consumed in big endian order by an ongoing SHA-256 evaluation.
```

Comment:
Prefer we drop the phrase "big endian order", since hashes don't have endianness, only conversions from hashes to numbers have endianness. You can say "without reversing it" if you want to emphasize that there is no satoshi crap going on with this function.

Reply:
Done.